### PR TITLE
Release checklist updated

### PR DESCRIPTION
**Pull request by biafra23**
Changes proposed in this pull request:
- changed errors
- improved wording
- clarified steps

@gnosis/mobile-devs

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/release_new_version.md [hunk 1]
```diff
@@ -25,10 +25,12 @@ git push -u origin release
   - [ ] Tag release using `vX.Y.Z` format
```
[the remark this answers is not in the document]
Comment (elgatovital) [reply .github/ISSUE_TEMPLATE/release_new_version.md:25]: we could be also more specific and rename "Draft new release" to "Draft new release in GitHub"

[the remark this answers is not in the document]
Comment (biafra23) [reply .github/ISSUE_TEMPLATE/release_new_version.md:25]: Weirdly there was no UI on Github to get to the release page. I had to enter the URL from memory.

[the remark this answers is not in the document]
Comment (biafra23) [reply .github/ISSUE_TEMPLATE/release_new_version.md:25]: >  Select release branch as target
Isn't this covered by this line: 
image

[the remark this answers is not in the document]
Comment (elgatovital) [reply .github/ISSUE_TEMPLATE/release_new_version.md]: it's a different step. When drafting a release master is selected by default, so you have to explicitly select release branch.

File: .github/ISSUE_TEMPLATE/release_new_version.md [hunk 2]
```diff
@@ -21,14 +21,18 @@ git push -u origin release
 
 **AFTER PRODUCT OWNER APPROVAL**
 
-- [ ] Draft new release
-  - [ ] Tag release using `vX.Y.Z` format
+- [ ] Create a Tag on the release branch with the release version using the `vX.Y.Z` format
```
Comment (elgatovital) [inline .github/ISSUE_TEMPLATE/release_new_version.md:24]: please revert to initial version. Tag will be created when drafting the release and should not be pushed manually. Let's not expose such implementation details.

File: app/google-services-debug-2023.json [hunk 3]
```diff
@@ -0,0 +1,231 @@
+{
```
Comment (elgatovital) [inline app/google-services-debug-2023.json:1]: This should not be added to the repo.

**Review by elgatovital — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/release_new_version.md (hunk 2 above)
Comment (biafra23) [reply .github/ISSUE_TEMPLATE/release_new_version.md]: I rephrased it. So it is clear that the tag will be created when the release is published.

**Review by elgatovital — APPROVED** [review]
👍

**Pull request merged**